docs/mdx-plugins/remark-link-rewrite.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path');
+const visit = require('unist-util-visit');
+const { URL } = require('url');
+
+const DEFAULT_OPTIONS = {
+  extension: 'md',
+  pagesDir: 'pages',
+  trailingSlash: true,
+};
+
+/**
+ * This rewrites internal MDX links to absolute URLs from the root domain.
+ * It's the similar behavior of GitHub MDX linking, but for Next.
+ *
+ * @param {object} options
+ * @param {string} [options.extension="md"]
+ * @param {string} [options.pagesDir="pages"]
+ * @param {boolean} [options.trailingSlash=true]
+ * @returns {function} remark plugin
+ */
+module.exports = function remarkLinkRewrite(options) {
+  const settings = Object.assign({}, DEFAULT_OPTIONS, options);
```

Comment:
Hmm, why not to go with the `newer` approach? 🤔 
```suggestion
  const settings = { ...DEFAULT_OPTIONS, ...options };
```

Reply:
You are right, I just copied this from an example `remark-` plugin.